File: gnn_model/data/transaction_graph.py
from .account import Account
from .transfer import Transfer

class TransactionGraph:
    def __init__(self, nodes: list[Account], edges: list[Transfer]):
        self.nodes: list[Account] = nodes
        self.edges = edges
    
    @classmethod
    def from_json(cls, json_data: list[dict]):
        nodes = list(map(lambda x: Account.from_json(x), json_data['nodes']))
        edges = list(map(lambda x: Transfer.from_json(x), json_data['edges']))
        return cls(nodes, edges)

File: graphaml/src/main/java/ru/uniserg/graphaml/data/TransactionGraph.java
package ru.uniserg.graphaml.data;

import java.util.List;

public class TransactionGraph {
    private List<Account> nodes;
    private List<Transaction> edges;

    public List<Account> getNodes() {
        return nodes;
    }

    public void setNodes(List<Account> nodes) {
        this.nodes = nodes;
    }

    public List<Transaction> getEdges() {
        return edges;
    }

    public void setEdges(List<Transaction> edges) {
        this.edges = edges;
    }

    @Override
    public String toString() {
        return "TransactionGraph{" +
                "nodes=" + nodes +
                ", edges=" + edges +
                '}';
    }
}
